stko/optimizers/collapser.py

```diff
@@ -0,0 +1,292 @@
+            the calculation are written, if ``None`` then
+            :func:`uuid.uuid4` is used.
+
+        step_size : :class:`float`
+            The relative size of the step to take during collapse.
+
+        distance_cut : :class:`float`
+            Distance between distinct building blocks to use as
+            threshold for halting collapse in Angstrom.
+
+        scale_steps : :class:`bool`, optional
+            Whether to scale the step of each distict building block
+            by their relative distance from the molecules centroid.
+            Defaults to ``True``
+
+        """
+
+        self._output_dir = output_dir
+        self._step_size = step_size
+        self._distance_cut = distance_cut
+        self._scale_steps = scale_steps
+
+    def _get_inter_BB_distance(self, mol):
+        """
+        Yield inter building block distances.
+
+        Ignores H atoms.
+
+        """
+
+        position_matrix = mol.get_position_matrix()
+
+        for atom1, atom2 in combinations(mol.get_atom_infos(), 2):
+            chk1 = (
+                atom1.get_atom().get_id() != atom2.get_atom().get_id()
+            )
+            chk2 = (
+                atom1.get_atom().get_atomic_number() != 1
+                and atom2.get_atom().get_atomic_number() != 1
+            )
+            chk3 = (
+                atom1.get_building_block_id() !=
+                atom2.get_building_block_id()
+            )
+            if chk1 and chk2 and chk3:
+                dist = get_atom_distance(
+                    position_matrix=position_matrix,
+                    atom1_id=atom1.get_atom().get_id(),
+                    atom2_id=atom2.get_atom().get_id()
+                )
+                yield dist
+
+    def _get_min_inter_BB_distance(self, mol):
+        """
+        Calculate the minimum inter building block distance.
+
+        """
+
+        min_dist = 1000
+        for dist in self._get_inter_BB_distance(mol):
+            min_dist = min([dist, min_dist])
+
+        return min_dist
+
+    def _no_short_contacts(self, mol):
+        """
+        Define a function that determines when collapsing should stop.
+
+        """
+
+        for dist in self._get_inter_BB_distance(mol):
+            if dist < self._distance_cut:
+                return False
+
+        return True
+
+    def _get_new_position_matrix(self, mol, step, vectors, scales):
+        """
+        Get the position matrix of the mol after translation.
+
+        """
+
+        new_position_matrix = mol.get_position_matrix()
+        for atom in mol.get_atom_infos():
+            bb_id = atom.get_building_block_id()
+            id = atom.get_atom().get_id()
+            pos = mol.get_position_matrix()[id]
+            new_position_matrix[id] = (
+                pos - step*vectors[bb_id]*scales[bb_id]
+            )
+
+        return new_position_matrix
+
+    def _get_BB_vectors(self, mol, BB_atom_ids):
```

Comment:
I don't understand why some methods have docs but others don't?

Reply:
Not every function requires docs is the main reason

Reply:
That said, some of these doc strings could improve and have been improved